### PR TITLE
[V2] test: add and edit unit tests to replica.go

### PR DESCRIPTION
**Pull request by hccheng72**
**What type of PR is this?**
/kind test

**What this PR does / why we need it**:
Add and edit unit tests to `pkg/controller/replica.go`

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/controller/replica_test.go [hunk 1]
```diff
@@ -68,7 +69,7 @@ func TestReplicaReconcile(t *testing.T) {
 				replicaAttachment := testReplicaAzVolumeAttachment
 				now := metav1.Time{Time: metav1.Now().Add(-1000)}
 				replicaAttachment.DeletionTimestamp = &now
-				replicaAttachment.Status.State = azdiskv1beta2.Detaching
+				replicaAttachment.Status.State = azdiskv1beta2.DetachmentFailed
```
Comment (sunpa93) [inline pkg/controller/replica_test.go:72]: Let's remove this line for this test case, attachment's state should not have any impact on this test case and will only confuse the readers going forward.

Instead, consider adding a new test case "[Success] Should delete replica attachment if attachment fails and requeue" to ensure that the reconciler will delete replica attachment in `AttachmentFailed` state and return reconcile result, where `requeue == true`

Comment (sunpa93) [reply pkg/controller/replica_test.go:72]: >Instead, consider adding a new test case "[Success] Should delete replica attachment if attachment fails and requeue" to ensure that the reconciler will delete replica attachment in AttachmentFailed state and return reconcile result, where requeue == true

Oh, you have already added "[Success] Should delete a failed-attachment replica." Let's go ahead and remove this line in that case 👍

File: pkg/controller/replica_test.go [hunk 2]
```diff
@@ -114,6 +115,80 @@ func TestReplicaReconcile(t *testing.T) {
...
+					replicaAttachment)
+
+				mockClients(controller.cachedClient.(*mockclient.MockClient), controller.azClient, controller.kubeClient)
+				return controller
+			},
+			verifyFunc: func(t *testing.T, controller *ReconcileReplica, result reconcile.Result, err error) {
+				require.NoError(t, err)
+				require.False(t, result.Requeue)
+
+				_, err = controller.azClient.DiskV1beta2().AzVolumeAttachments(testNamespace).Get(context.TODO(), testPersistentVolume0Name+"-"+testNode1Name+"-attachment", metav1.GetOptions{})
```
Comment (sunpa93) [inline pkg/controller/replica_test.go:188]: Use `testReplicaAzVolumeAttachment.Name()` instead.

File: pkg/controller/replica_test.go (hunk 1 above)
Comment (hccheng72) [reply pkg/controller/replica_test.go:72]: I edited the attachment's state to `DetachmentFailed` for testing [this case](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/fde668a0d77effb4af845a115572180f85a65f9f/pkg/controller/replica.go#L95). And we can verify if the state has been updated to `ForceDetachPending` right before [delete the attachment manually in the verifyFun](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/fde668a0d77effb4af845a115572180f85a65f9f/pkg/controller/replica_test.go#L96). Not sure if it's needed?

Comment (sunpa93) [reply pkg/controller/replica_test.go:72]: For this test case, we want to check if a new replica attachment is created when one is deleted.

Comment (hccheng72) [reply pkg/controller/replica_test.go]: Got it. I've updated the code.

**Review by sunpa93 — APPROVED** [review]
/lgtm

**Pull request merged**